src/Microsoft.OData.Core/SelectedPropertiesNode.cs

```diff
@@ -81,6 +87,14 @@ internal SelectedPropertiesNode(string selectClause)
                 string[] segments = t.Split(PathSeparator);
                 this.ParsePathSegment(segments, 0);
             }
+
+            if (this.selectedProperties != null
+                && this.selectedProperties.Count > 0
+                && this.selectedProperties.All(IsValidExpandToken))
+            {
+                // select list is not empty and only contains expand tokens
+                this.selectionType = SelectionType.EntireSubtree;

```

Comment:
>this.selectionType = SelectionType.EntireSubtree; [](start = 16, length = 49)

Line 76: call private constructor to set "selectionType" as PartialSubtree, meantime, this constructor reset it EntireSubtree if meet some condition. 

So, in this case, why don't you remove the private constructor and move the content in this constructor?

Reply:
If that's not the case, i think the line 90-97 should move out of the constructor into the static Create(...) method?

---
In reply to: [221684657](https://github.com/OData/odata.net/pull/1264#discussion_r221684657) [](ancestors = 221684657)

Reply:
+1

---
In reply to: [221684657](https://github.com/OData/odata.net/pull/1264#discussion_r221684657) [](ancestors = 221684657)